### PR TITLE
[NEAT-563, NEAT-564, NEAT-565] 	🐉 Simplify classic extractors

### PR DESCRIPTION
**Pull request by doctrino**
... and add some new functionality.

[NEAT-563] Json inference was not a bug, but an issue with how the test data was set up.

[NEAT-563]: https://cognitedata.atlassian.net/browse/NEAT-563?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/tests_unit/graph/test_extractors/test_timeseries_extractor.py [hunk 1]
```diff
@@ -17,4 +17,4 @@ def test_timeseries_extractor():
     for triple in TimeSeriesExtractor.from_dataset(client_mock, data_set_external_id="some data set").extract():
         g.add(triple)
 
-    assert len(g) == 20
+    assert len(g) == 22
```
Comment (doctrino) [inline tests/tests_unit/graph/test_extractors/test_timeseries_extractor.py:20]: Bug in last one. `isString` and `isSteg` was only counted if `True`.

File: tests/tests_unit/graph/test_extractors/test_sequences_extractor.py [hunk 2]
```diff
@@ -17,4 +17,4 @@ def test_sequences_extractor():
     for triple in SequencesExtractor.from_dataset(client_mock, data_set_external_id="some data set").extract():
         g.add(triple)
 
-    assert len(g) == 14
+    assert len(g) == 16
```
Comment (doctrino) [inline tests/tests_unit/graph/test_extractors/test_sequences_extractor.py:20]: Fix in last one. `Columns` were not included.

File: tests/tests_unit/rules/test_importers/test_inference_importer.py [hunk 3]
```diff
@@ -72,7 +72,7 @@ def test_json_value_type_inference():
 
     properties = {prop.property_: prop for prop in rules.properties}
 
-    assert len(rules.properties) == 9
+    assert len(rules.properties) == 8
```
Comment (doctrino) [inline tests/tests_unit/rules/test_importers/test_inference_importer.py:75]: Now we are no longer getting `externalId`

File: cognite/neat/_graph/extractors/_classic_cdf/_base.py [hunk 4]
```diff
@@ -104,9 +116,48 @@ def extract(self) -> Iterable[Triple]:
...
+        dumped.pop("parentExternalId", None)
+        if "metadata" in dumped:
+            triples.extend(self._metadata_to_triples(id_, dumped.pop("metadata")))
+        if "columns" in dumped:
+            columns = dumped.pop("columns")
+            triples.append(
+                (id_, self.namespace.columns, Literal(json.dumps({"columns": columns}), datatype=XSD._NS["json"]))
+            )
+
+        for key, value in dumped.items():
```
Comment (nikokaoja) [inline cognite/neat/_graph/extractors/_classic_cdf/_base.py:148]: here we are sure that value is not list of values

File: cognite/neat/_graph/transformers/_classic_cdf.py [hunk 5]
```diff
@@ -75,7 +77,31 @@ def get_depth(
             return None
 
 
-class AssetTimeSeriesConnector(BaseTransformer):
+class ClassicConnector(BaseTransformer, ABC):
```
Comment (nikokaoja) [inline cognite/neat/_graph/transformers/_classic_cdf.py:80]: `BaseAssetConnector` better name?

File: cognite/neat/_graph/transformers/_classic_cdf.py [hunk 6]
```diff
@@ -75,7 +77,31 @@ def get_depth(
...
+        self._attribute = attribute or self._default_attribute
+
+    def transform(self, graph: Graph) -> None:
+        for item_id, *_ in graph.query(self._select_item_ids.format(item_type=self._item_type)):  # type: ignore[misc]
+            for asset_id, *_ in graph.query(  # type: ignore[misc]
+                self._select_connected_assets.format(
+                    item_id=item_id, attribute=self._attribute, asset_type=self._asset_type
+                )
+            ):
+                graph.add((asset_id, self._connection_type, item_id))
```
Comment (nikokaoja) [inline cognite/neat/_graph/transformers/_classic_cdf.py:101]: use `add_triples_in_batch` to batch write triples instead individually writing them

**Review by nikokaoja — APPROVED** [review]
Nice clean up!

File: cognite/neat/_graph/extractors/_classic_cdf/_base.py (hunk 4 above)
Comment (doctrino) [reply cognite/neat/_graph/extractors/_classic_cdf/_base.py:148]: Yes that is handled in two lines below.

File: cognite/neat/_graph/transformers/_classic_cdf.py (hunk 5 above)
Comment (doctrino) [reply cognite/neat/_graph/transformers/_classic_cdf.py:80]: agreed

**Pull request merged**